.travis.yml

```diff
@@ -42,25 +43,46 @@ matrix:
     - php: 7.3
       env:
         - DEPS=latest
-
-services: mongodb
+    - php: 7.4
+      env:
+        - DEPS=lowest
+    - php: 7.4
+      env:
+        - DEPS=locked
+    - php: 7.4
+      env:
+        - DEPS=latest
 
 before_install:
   - travis_retry composer self-update
+  - shopt -s expand_aliases
   - if [[ $DEPS == 'lowest' ]]; then travis_retry composer remove $COMPOSER_ARGS --no-update --dev $LOWEST_DEPS_REMOVE ; fi
+  - alias composer="travis_retry docker-compose run --rm php composer"
+  - alias php="travis_retry docker-compose run -e TRAVIS -e TRAVIS_JOB_ID --rm php php"
+  - docker-compose build --build-arg PHP_VERSION=${TRAVIS_PHP_VERSION} --build-arg XDEBUG=${TEST_COVERAGE:+1} --no-cache php
```

Comment:
This is by far the longest step… could be reduced in the future if we had a set of images we would maintain and push on docker hub.